src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryBlockOpen.kt

```diff
@@ -26,27 +28,41 @@ object ChocolateFactoryBlockOpen {
      */
     private val commandPattern by RepoPattern.pattern(
         "inventory.chocolatefactory.opencommand",
-        "\\/(?:cf|(?:chocolate)?factory)(?: .*)?",
+        "/(?:cf|(?:chocolate)?factory)(?: .*)?",
     )
 
     private var commandSentTimer = SimpleTimeMark.farPast()
 
     @SubscribeEvent
     fun onCommandSend(event: MessageSendToServerEvent) {
-        if (!isEnabled()) return
+        if (!LorenzUtils.inSkyBlock) return
         if (!commandPattern.matches(event.message)) return
         if (commandSentTimer.passedSince() < 5.seconds) return
-        if (MythicRabbitPetWarning.correctPet()) return
 
-        commandSentTimer = SimpleTimeMark.now()
-        event.cancel()
-        ChatUtils.clickToActionOrDisable(
-            "§cBlocked opening the Chocolate Factory without a §dMythic Rabbit Pet §cequipped!",
-            config::mythicRabbitRequirement,
-            actionName = "open pets menu",
-            action = { HypixelCommands.pet() },
-        )
+        if (config.mythicRabbitRequirement && !MythicRabbitPetWarning.correctPet()) {
+            event.cancelOpen()
+            ChatUtils.clickToActionOrDisable(
+                "§cBlocked opening the Chocolate Factory without a §dMythic Rabbit Pet §cequipped!",
+                config::mythicRabbitRequirement,
+                actionName = "open pets menu",
+                action = { HypixelCommands.pet() },
+            )
+        } else if (config.boosterCookieRequirement) {
+            profileStorage?.boosterCookieExpiryTime?.let {
+                if (it.timeUntil() > 0.seconds) return
+                event.cancelOpen()
+                ChatUtils.clickToActionOrDisable(
+                    "§cBlocked opening the Chocolate Factory without a §dBooster Cookie §cactive!",
+                    config::boosterCookieRequirement,
+                    actionName = "open bazaar",
+                    action = { HypixelCommands.bazaar("Booster Cookie") },
```

Comment:
i dont think this will work without a  booster cookie. how about warping to hub and starting a lovely pathfind to the bazaar npc?